### PR TITLE
Proofread SAM, Selective Backprop, Sequential Length Warmup, and Squeeze Excite Docstrings

### PR DESCRIPTION
**Pull request by mcneela**
Proofread SAM, Selective Backprop, Sequential Length Warmup, and Squeeze Excite Docstrings

### CODE REVIEW

File: composer/algorithms/scale_schedule/scale_schedule.py [hunk 1]
```diff
@@ -32,11 +31,11 @@ def __init__(self, ratio: float = 1.0):
             category=DeprecationWarning)
 
     def match(self, event: Event, state: State) -> bool:
-        """Run on Event.INIT.
+        """Run on :attr:`~composer.core.event.Event.INIT`.
```
Comment (hanlint) [inline composer/algorithms/scale_schedule/scale_schedule.py:34]: can you try :attr:`~Event.INIT`? That should work, and we want to the docstrings readable.

File: composer/algorithms/scale_schedule/scale_schedule.py [hunk 2]
```diff
@@ -32,11 +31,11 @@ def __init__(self, ratio: float = 1.0):
...
 
         Args:
-            event (:class:`Event`): The current event.
-            state (:class:`State`): The current state.
+            event (:class:`~composer.core.event.Event`): The current event.
```
Comment (hanlint) [inline composer/algorithms/scale_schedule/scale_schedule.py:37]: Same as above comment, try :class:`~Event`.

**Review by hanlint — CHANGES_REQUESTED** [review]
One requested change on using the ~ shorthand

File: composer/algorithms/scale_schedule/scale_schedule.py (hunk 1 above)
Comment (mcneela) [reply composer/algorithms/scale_schedule/scale_schedule.py:34]: Gotcha.

Comment (mcneela) [reply composer/algorithms/scale_schedule/scale_schedule.py:34]: @hanlint I just tried this, and it doesn't link to the class with the shortened syntax. :(

**Review by hanlint — APPROVED** [review]

File: composer/algorithms/sam/sam.py [hunk 3]
```diff
@@ -20,14 +20,25 @@
...
 class SAMOptimizer(torch.optim.Optimizer):
-    """Wraps an optimizer with sharpness-aware minimization (`Foret et al, 2020 <https://arxiv.org/abs/2010.01412>`_).
+    """Wraps an optimizer with sharpness-aware minimization 
+    (`Foret et al, 2020 <https://arxiv.org/abs/2010.01412>`_).
     See :class:`SAM` for details.
 
     Implementation based on https://github.com/davda54/sam
+
+    Args:
+        base_optimizer (:class:`torch.optim.Optimizer`) The optimizer to apply SAM to.
```
Comment (ravi-mosaicml) [inline composer/algorithms/sam/sam.py:30]:
```suggestion
        base_optimizer (torch.optim.Optimizer) The optimizer to apply SAM to.
```

File: composer/algorithms/sam/sam.py [hunk 4]
```diff
@@ -20,14 +20,25 @@
 
 
 class SAMOptimizer(torch.optim.Optimizer):
-    """Wraps an optimizer with sharpness-aware minimization (`Foret et al, 2020 <https://arxiv.org/abs/2010.01412>`_).
+    """Wraps an optimizer with sharpness-aware minimization
```
Comment (ravi-mosaicml) [inline composer/algorithms/sam/sam.py:23]:
```suggestion
    """Wraps an optimizer with sharpness-aware minimization (`Foret et al, 2020 <https://arxiv.org/abs/2010.01412>`_).
```

File: composer/algorithms/sam/sam.py [hunk 5]
```diff
@@ -20,14 +20,25 @@
 
 
 class SAMOptimizer(torch.optim.Optimizer):
-    """Wraps an optimizer with sharpness-aware minimization (`Foret et al, 2020 <https://arxiv.org/abs/2010.01412>`_).
+    """Wraps an optimizer with sharpness-aware minimization 
+    (`Foret et al, 2020 <https://arxiv.org/abs/2010.01412>`_).
```
Comment (ravi-mosaicml) [inline composer/algorithms/sam/sam.py:24]:
```suggestion

```

File: composer/algorithms/selective_backprop/selective_backprop.py [hunk 6]
```diff
@@ -73,9 +73,9 @@ def select_using_loss(input: torch.Tensor,
     will still be used for the weight gradient computation.
 
     Args:
-        input (torch.Tensor): Input tensor to prune
-        target (torch.Tensor): Output tensor to prune
-        model (Callable): Model with which to predict outputs
+        input (:class:`torch.Tensor`): Input tensor to prune.
```
Comment (ravi-mosaicml) [inline composer/algorithms/selective_backprop/selective_backprop.py:76]:
```suggestion
        input (torch.Tensor): Input tensor to prune.
```

File: composer/algorithms/selective_backprop/selective_backprop.py [hunk 7]
```diff
@@ -73,9 +73,9 @@ def select_using_loss(input: torch.Tensor,
...
+        target (:class:`torch.Tensor`): Output tensor to prune.
```
Comment (ravi-mosaicml) [inline composer/algorithms/selective_backprop/selective_backprop.py:77]:
```suggestion
        target (torch.Tensor): Output tensor to prune.
```

File: composer/algorithms/squeeze_excite/squeeze_excite.py [hunk 8]
```diff
@@ -27,23 +27,23 @@ def apply_squeeze_excite(
...
     feeds the resulting vector to a single-hidden-layer fully-connected
-    network (MLP), and uses the output of this MLP as attention coefficients
+    network (MLP), and uses the outputs of this MLP as attention coefficients
     to rescale the input. This allows the network to take into account global
     information about each input, as opposed to only local receptive fields
     like in a convolutional layer.
 
     Args:
-        model (torch.nn.Module): The module to apply squeeze excite replacement.
+        model (:class:`torch.nn.Module`): The module to apply squeeze excite replacement to.
```
Comment (ravi-mosaicml) [inline composer/algorithms/squeeze_excite/squeeze_excite.py:36]:
```suggestion
        model (torch.nn.Module): The module to apply squeeze excite replacement to.
```

File: composer/algorithms/squeeze_excite/squeeze_excite.py [hunk 9]
```diff
@@ -27,23 +27,23 @@ def apply_squeeze_excite(
...
             MLP. If less than 1, interpreted as a fraction of the number of
             output channels in the :class:`~torch.nn.Conv2d` immediately
             preceding each Squeeze-and-Excitation block. Default: ``64``.
         min_channels (int, optional): An SE block is added after a :class:`~torch.nn.Conv2d`
             module ``conv`` only if one of the layer's input or output channels is greater than
             this threshold. Default: ``128``.
-        optimizers (torch.optim.Optimizer | Sequence[torch.optim.Optimizer], optional):
-            Existing optimizers bound to ``model.parameters()``. All optimizers that have already been
-            constructed with ``model.parameters()`` must be specified here so
+        optimizers (:class:`torch.optim.Optimizer` | Sequence[:class:`torch.optim.Optimizer`], optional):
```
Comment (ravi-mosaicml) [inline composer/algorithms/squeeze_excite/squeeze_excite.py:44]:
```suggestion
        optimizers (torch.optim.Optimizer | Sequence[torch.optim.Optimizer], optional):
```

File: composer/algorithms/squeeze_excite/squeeze_excite.py [hunk 10]
```diff
@@ -154,8 +157,8 @@ def match(self, event: Event, state: State) -> bool:
         """Runs on :attr:`~composer.core.event.Event.INIT`
 
         Args:
-            event (Event): The current event.
-            state (State): The current state.
+            event (:class:`~composer.core.event.Event`): The current event.
```
Comment (ravi-mosaicml) [inline composer/algorithms/squeeze_excite/squeeze_excite.py:160]:
```suggestion
            event (Event): The current event.
```

File: composer/algorithms/squeeze_excite/squeeze_excite.py [hunk 11]
```diff
@@ -154,8 +157,8 @@ def match(self, event: Event, state: State) -> bool:
...
+            state (:class:`~composer.core.state.State`): The current state.
```
Comment (ravi-mosaicml) [inline composer/algorithms/squeeze_excite/squeeze_excite.py:161]:
```suggestion
            state (State): The current state.
```

File: composer/algorithms/squeeze_excite/squeeze_excite.py [hunk 12]
```diff
@@ -165,9 +168,9 @@ def apply(self, event: Event, state: State, logger: Logger) -> Optional[int]:
         """Applies the Squeeze-and-Excitation layer replacement.
 
         Args:
-            event (Event): the current event
-            state (State): the current trainer state
-            logger (Logger): the training logger
+            event (:class:`~composer.core.event.Event`): The current event.
```
Comment (ravi-mosaicml) [inline composer/algorithms/squeeze_excite/squeeze_excite.py:171]:
```suggestion
            event (Event): The current event.
```

File: composer/algorithms/squeeze_excite/squeeze_excite.py [hunk 13]
```diff
@@ -165,9 +168,9 @@ def apply(self, event: Event, state: State, logger: Logger) -> Optional[int]:
...
+            state (:class:`~composer.core.state.State`): The current trainer state.
```
Comment (ravi-mosaicml) [inline composer/algorithms/squeeze_excite/squeeze_excite.py:172]:
```suggestion
            state (State): The current trainer state.
```

File: composer/algorithms/squeeze_excite/squeeze_excite.py [hunk 14]
```diff
@@ -165,9 +168,9 @@ def apply(self, event: Event, state: State, logger: Logger) -> Optional[int]:
...
+            logger (:class:`~composer.loggers.Logger`): The training logger.
```
Comment (ravi-mosaicml) [inline composer/algorithms/squeeze_excite/squeeze_excite.py:173]:
```suggestion
            logger (Logger): The training logger.
```

**Review by ravi-mosaicml — APPROVED** [review]
LGTM
